src/components/fields/image-upload/image-review/image-review.tsx

```diff
@@ -117,30 +119,36 @@ export const ImageReview = (props: IProps) => {
 	// =============================================================================
 	// - REVIEW MODAL
 	// =============================================================================
-	const handleSelectFile = (selectedFile: File) => {
+	const handleSelectFile = (selectedFiles: File[]) => {
 		if (
 			!maxFiles ||
-			(selectedFile &&
-				images.filter(({ addedFrom, status }) => status >= EImageStatus.NONE || addedFrom === "reviewModal")
-					.length < maxFiles)
+			selectedFiles.length +
+				images.filter(({ status, addedFrom }) => status >= EImageStatus.NONE || addedFrom === "reviewModal")
+					.length <=
+				maxFiles
 		) {
-			// image manager will handle the rest
-			setImages((prev) => {
-				const slot = ImageUploadHelper.findAvailableSlot(prev);
-				return [
-					...prev,
-					{
-						id: generateRandomId(),
-						file: selectedFile,
-						name: selectedFile.name,
-						dimensions,
-						status: EImageStatus.NONE,
-						uploadProgress: 0,
-						addedFrom: "reviewModal",
-						slot,
-					},
-				];
+			selectedFiles.forEach((selectedFile) => {
+				setImages((prev) => {
+					const slot = ImageUploadHelper.findAvailableSlot(prev);
+					return [
+						...prev,
+						{
+							id: generateRandomId(),
+							file: selectedFile,
+							name: selectedFile.name,
+							dimensions,
+							status: EImageStatus.NONE,
+							uploadProgress: 0,
+							addedFrom: "reviewModal",
+							slot,
+						},
+					];
+				});
 			});
+		} else {
+			setImages(images.filter(({ status }) => status == EImageStatus.UPLOADED));
+			setExceedError(true);
+			onExit();
```

Comment:
it's rather abrupt to close the review modal, perhaps we can have the review modal show an error message instead? (it's already showing error messages for unsupported photo formats or exceeded file size), may I trouble you to check with UX on this?